rpcserver.go

```diff
@@ -456,6 +460,83 @@ func determineFeePerKw(feeEstimator lnwallet.FeeEstimator, targetConf int32,
 	}
 }
 
+func (r *rpcServer) ListUnspent(ctx context.Context,
+	in *lnrpc.ListUnspentRequest) (*lnrpc.ListUnspentResponse, error) {
+	minConfs := in.Minconfs
+	maxConfs := in.Maxconfs
+	if minConfs >= maxConfs {
+		return nil, fmt.Errorf("Max confirmations must be > min " +
+			"confirmations")
+	}
+	usingMax := maxConfs != math.MaxInt32
+	if usingMax {
+		rpcsLog.Infof("[listunspent] min=%v max=%v", minConfs, maxConfs)
+	} else {
+		rpcsLog.Infof("[listunspent] min=%v", minConfs)
+	}
+
+	utxos1, err := r.server.cc.wallet.ListUnspentWitness(minConfs)
+	if err != nil {
+		return nil, err
+	}
+
+	utxos := make([]*lnwallet.Utxo, 0, len(utxos1))
+
+	if usingMax {
+		utxos2, err := r.server.cc.wallet.ListUnspentWitness(maxConfs)
+		if err != nil {
+			return nil, err
+		}
+		// We want the symmetric difference between the first set (utxos1)
+		// and the second (utxos2) (because utxos2 is a subset), and don't want
+		// to import a set package, also don't want most naive case (scales
+		// quadratically), make a map of utxo2 and then key utxo1 into it:
+		utxo2map := make(map[string]*lnwallet.Utxo, len(utxos2))
+		for _, u := range utxos2 {
+			utxo2map[u.OutPoint.String()] = u
+		}
+
+		for _, u := range utxos1 {
+			if _, ok := utxo2map[u.OutPoint.String()]; !ok {
+				utxos = append(utxos, u)
+			}
+		}
+	} else {
+		utxos = utxos1
+	}
+
+	utxoresplist := []*lnrpc.Utxo{}
+	for _, utxo := range utxos {
+		utxoresp := lnrpc.Utxo{
+			Amount:       int64(utxo.Value),
+			Scriptpubkey: hex.EncodeToString(utxo.PkScript),
+			Outpoint:     utxo.OutPoint.String(),
+		}
+		_, outAddresses, _, err := txscript.ExtractPkScriptAddrs(
+			utxo.PkScript, activeNetParams.Params)
+		if err != nil {
+			return nil, err
+		}
+		if len(outAddresses) != 1 {
+			return nil, fmt.Errorf("An output was unexpectedly multisig")
+		}
+		utxoresp.Address = outAddresses[0].String()
+		if utxo.AddressType == 0 {
+			utxoresp.AddressType = "p2wkh"
```

Comment:
as @joostjager mentioned, here we should be using the already defined enums.

Reply:
Done, but have Qs about this, see comment below